scripts/generate-authors.js

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env node
+
+/*
+ * Generate an AUTHOR file on the repo root and on each lerna package based on git log.
+ *
+ * Add / change the ALIASES map to avoid duplications and show the correct names / emails.
+ *
+ * NOTE: Author lines with `users.noreply.github.com` emails are removed.
+ */
+
+const ALIASES = {
+  'Anna Henningsen <anna.henningsen@mongodb.com>': 'Anna Henningsen <addaleax@gmail.com>',
+  'Anna Henningsen <anna@addaleax.net>': 'Anna Henningsen <addaleax@gmail.com>',
+  'Anna Henningsen <github@addaleax.net>': 'Anna Henningsen <addaleax@gmail.com>',
+  'aherlihy <anna.herlihy@10gen.com>': 'Anna Herlihy <herlihyap@gmail.com>',
+  'anna herlihy <anna.herlihy@10gen.com>': 'Anna Herlihy <herlihyap@gmail.com>',
+  'Massimiliano Marcon <max.marcon@mongodb.com>': 'Massimiliano Marcon <me@marcon.me>',
+  'mcasimir <maurizio.cas@gmail.com>': 'Maurizio Casimirri <maurizio.cas@gmail.com>',
+}
+
+const { execSync } = require('child_process');
+const path = require('path');
+const fs = require('fs');
+
+const packageRootPath = path.resolve(__dirname, '..');
+
+function getAuthorsGitLog(packagePath) {
+  return execSync(
+    `bash -c "git log --format='%aN <%aE>' -- ${packagePath} | grep -v "users.noreply.github.com" | sort -f | uniq"`,
```

Comment:
Fwiw, if you want to use aliases, you can use `git log --use-mailmap` and then use a `.mailmap` file as a standardized format that git understands for mapping aliases, like https://github.com/nodejs/node/blob/master/.mailmap

Reply:
Much better